src/Response.php

```diff
@@ -46,21 +85,87 @@ public function writeContinue()
         $this->conn->write("HTTP/1.1 100 Continue\r\n\r\n");
     }
 
+    /**
+     * Writes the given HTTP message header.
+     *
+     * This method MUST be invoked once before calling `write()` or `end()` to send
+     * the actual HTTP message body:
+     *
+     * ```php
+     * $response->writeHead(200, array(
+     *     'Content-Type' => 'text/plain'
+     * ));
+     * $response->end('Hello World!');
+     * ```
+     *
+     * Calling this method more than once will result in an `Exception`.
+     *
+     * Unless you specify a `Content-Length` header yourself, the response message
+     * will automatically use chunked transfer encoding and send the respective header
+     * (`Transfer-Encoding: chunked`) automatically. If you know the length of your
+     * body, you MAY specify it like this instead:
+     *
+     * ```php
+     * $data = 'Hello World!';
+     *
+     * $response->writeHead(200, array(
+     *     'Content-Type' => 'text/plain',
+     *     'Content-Length' => strlen($data)
+     * ));
+     * $response->end($data);
+     * ```
+     *
+     * Note that it will automatically assume a `X-Powered-By: react/alpha` header
+     * unless your specify a custom `X-Powered-By` header yourself:
+     *
+     * ```php
+     * $response->writeHead(200, array(
+     *     'X-Powered-By' => 'PHP 3'
+     * ));
+     * ```
+     *
+     * If you do not want to send this header at all, you can use an empty array as
+     * value like this:
+     *
+     * ```php
+     * $response->writeHead(200, array(
+     *     'X-Powered-By' => array()
+     * ));
+     * ```
+     *
+     * @param int   $status
+     * @param array $headers
+     * @throws \Exception
+     */
     public function writeHead($status = 200, array $headers = array())
     {
         if ($this->headWritten) {
             throw new \Exception('Response head has already been written.');
         }
 
-        if (isset($headers['Content-Length'])) {
+        $lower = array_change_key_case($headers);
+
+        // disable chunked encoding if content-length is given
+        if (isset($lower['content-length'])) {
             $this->chunkedEncoding = false;
         }
 
-        $headers = array_merge(
-            array('X-Powered-By' => 'React/alpha'),
-            $headers
-        );
+        // assign default "X-Powered-By" header as first for history reasons
+        if (!isset($lower['x-powered-by'])) {
+            $headers = array_merge(
+                array('X-Powered-By' => 'React/alpha'),
+                $headers
+            );
+        }
```

Comment:
The `$lower` is only used within this method, so it's not needed anymore after this assignment.